c/polyjuice.h

```diff
@@ -299,36 +283,37 @@ int load_account_code(gw_context_t* gw_ctx, uint32_t account_id,
     return 0;
   }
 
-  uint64_t old_code_size = *code_size;
+  debug_print_int("[load_account_code] code_size before loading", *code_size);
   ret = gw_ctx->sys_load_data(gw_ctx, data_hash, code_size, offset, code);
+  debug_print_int("[load_account_code] code_size after loading", *code_size);
   if (ret != 0) {
     ckb_debug("[load_account_code] sys_load_data failed");
     return ret;
   }
-  if (*code_size > old_code_size) {
-    debug_print_int("[load_account_code] code can't be larger than", MAX_DATA_SIZE);
-    return -1;
+  if (*code_size > MAX_DATA_SIZE) {
+    debug_print_int("[load_account_code] code_size can't be larger than",
+                    MAX_DATA_SIZE);
+    return GW_FATAL_BUFFER_OVERFLOW;
   }
+
   return 0;
 }
 
-////////////////////////////////////////////////////////////////////////////
-//// Callbacks
-////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+//// Callbacks - EVMC Host Interfaces
+////////////////////////////////////////////////////////////////////////////////
 struct evmc_tx_context get_tx_context(struct evmc_host_context* context) {
   struct evmc_tx_context ctx{0};
+  memcpy(ctx.tx_origin.bytes, g_tx_origin.bytes, 20);
+
   /* gas price = 1 */
   ctx.tx_gas_price.bytes[31] = 0x01;
-  memcpy(ctx.tx_origin.bytes, g_tx_origin.bytes, 20);
-  uint8_t coinbase_script_hash[32] = {0};
-  int ret = context->gw_ctx->sys_get_script_hash_by_account_id(context->gw_ctx,
-                                                               context->gw_ctx->block_info.block_producer_id,
-                                                               coinbase_script_hash);
-  if (ret != 0) {
-    debug_print_int("get script hash by block producer id failed", context->gw_ctx->block_info.block_producer_id);
-    context->error_code = ret;
-  }
-  memcpy(ctx.block_coinbase.bytes, coinbase_script_hash, 20);
+
+  gw_reg_addr_t *block_producer = &context->gw_ctx->block_info.block_producer;
+  debug_print_data("load block_coinbase address:",
+                   block_producer->addr, ETH_ADDRESS_LEN);
+  memcpy(ctx.block_coinbase.bytes, block_producer->addr, ETH_ADDRESS_LEN);
```

Comment:
The length of block_producer->addr is not fixed to 20 bytes, we should check the length first, and return an error if it is invalid.

Reply:
https://github.com/nervosnetwork/godwoken-polyjuice/pull/138/commits/24cc37b1ea5687ae156351104006878c4ce2e018